### PR TITLE
Reduce 500s

### PR DESCRIPTION
**Pull request by neilalexander**
This reduces the number of 500s we return by:
- returning 404 on media download errors
- trying to avoid returning 500s on `/send` at all, since we're really supposed to just return a 200 with a list of errored event IDs where possible

500s are bad because they cause the remote server to back off and stop talking to us, which I've seen a lot in practice on `dendrite.neilalexander.dev`.

### CODE REVIEW

**Comment by Kegsay** [comment]
Please don't mess with `/send` too much, that file I'm hacking on currently.

**Comment by neilalexander** [comment]
@Kegsay That's all I plan to do - it's just enough to stop us returning a sea of 500s.

**Comment by Kegsay** [comment]
You can fix the merge conflicts now 😏

File: federationapi/routing/send.go [hunk 1]
```diff
@@ -135,22 +124,28 @@ func (t *txnReq) processTransaction() (*gomatrixserverlib.RespSend, error) {
 		}
 		if err := json.Unmarshal(pdu, &header); err != nil {
 			util.GetLogger(t.context).WithError(err).Warn("Transaction: Failed to extract room ID from event")
-			return nil, unmarshalError{err}
+			continue
```
Comment (Kegsay) [inline federationapi/routing/send.go:127]: We should probably just add an error to the `results` map like we do elsewhere (in which case just factor out the for loop to a new function which returns an `error` and set that if it's non-nil.

Comment (neilalexander) [reply federationapi/routing/send.go:127]: The difficulty here is that we don't know the event ID until after `gomatrixserverlib.NewEventFromUntrustedJSON`.

Comment (Kegsay) [reply federationapi/routing/send.go:127]: oh hah. Fair enough, might be worth adding a quick comment about that.

**Pull request merged**